inst/utils-0.0.1/addCountrySelect.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Adds a dropdown to highlight the selected country in the chart
+ *
+ * @param {Node} el - an element in the DOM
+ * @param {Array} results - KRI results (dfSummary)
+ * @param {Object} instance - Chart.js chart object
+ *
+ * @returns {Node} HTML select element
+ */
+
+
+// Add event listener to highlight countries.
+const addCountrySelect = function(el, results, instance, groupSelect) {
+
+    el.style.position = 'relative';
+
+    // add container in which to place dropdown
+    const countrySelectContainer = document.createElement('div');
+    el.appendChild(countrySelectContainer);
+
+    // position container absolutely in upper left corner of `el`.
+    countrySelectContainer.style.position = 'absolute';
+    countrySelectContainer.style.top = 0;
+    countrySelectContainer.style.left = "175px";
+
+    // add dropdown label
+    const countrySelectLabel = document.createElement('span');
+    countrySelectLabel.classList.add("gsm-country-select-label")
+    countrySelectLabel.innerHTML = "Country: ";
+    countrySelectContainer.appendChild(countrySelectLabel)
+
+    // add dropdown
+    const countrySelect = document.createElement('select');
+    countrySelectContainer.appendChild(countrySelect);
+
+    // add default option
+    const noneOption = document.createElement('option');
+    noneOption.innerHTML = 'None';
+    countrySelect.appendChild(noneOption);
+
+    // get sorted array of country
+    const countries = [
+        ...new Set(results.map(d => d.site.country))
```

Comment:
Grab the set of countries from `input.dfGroups` directly rather than from `input.dfSummary`.

Reply:
Hey @samussiah, one potential issue with this that I want to ask about. `input.`dfGroups` does not have a "GroupID" field within it. This isn't an issue with the lines you highlighted above, but that does mean in the example below, I need to hardcode `SiteID` as a mapped value. Is that okay?
![image](https://github.com/Gilead-BioStats/gsm/assets/76186904/2d10e1c0-b53d-4da4-99a6-c7aac9ee89f4)